File: packages/entity-extraction/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/entityextraction/classes/options/DownloadModelOptions.java
package io.capawesome.capacitorjs.plugins.mlkit.entityextraction.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.mlkit.entityextraction.EntityExtractionHelper;

public class DownloadModelOptions {

    private static final String ERROR_LANGUAGE_MISSING = "language must be provided.";

    @NonNull
    private final String modelIdentifier;

    public DownloadModelOptions(@NonNull PluginCall call) throws Exception {
        this.modelIdentifier = DownloadModelOptions.getModelIdentifierFromCall(call);
    }

    @NonNull
    public String getModelIdentifier() {
        return modelIdentifier;
    }

    @NonNull
    private static String getModelIdentifierFromCall(@NonNull PluginCall call) throws Exception {
        String language = call.getString("language");
        if (language == null) {
            throw new Exception(ERROR_LANGUAGE_MISSING);
        }
        return EntityExtractionHelper.convertLanguageToModelIdentifier(language);
    }
}

File: packages/entity-extraction/ios/Plugin/Classes/Options/DownloadModelOptions.swift
import Foundation
import Capacitor
import MLKitEntityExtraction

@objc public class DownloadModelOptions: NSObject {
    let modelIdentifier: EntityExtractionModelIdentifier

    init(_ call: CAPPluginCall) throws {
        self.modelIdentifier = try DownloadModelOptions.getModelIdentifierFromCall(call)
    }

    private static func getModelIdentifierFromCall(_ call: CAPPluginCall) throws -> EntityExtractionModelIdentifier {
        guard let language = call.getString("language") else {
            throw CustomError.languageMissing
        }
        return try EntityExtractionHelper.convertLanguageToModelIdentifier(language)
    }
}
